src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -146,16 +146,32 @@ def get_handler(id_or_link, project=None):
 
 def describe(id_or_link, **kwargs):
     '''
-    :param id_or_link: String containing an object ID or dict containing a DXLink
+    :param id_or_link: String containing an object ID or dict containing a DXLink,
+                       or a list of object ID's or dict's containing a DXLink.
 
     Given an object ID, calls :meth:`~dxpy.bindings.DXDataObject.describe` on the object.
 
     Example::
 
         describe("file-1234")
     '''
-    handler = get_handler(id_or_link)
-    return handler.describe(**kwargs)
+    # If this is a list, extract the ids.
+    if isinstance(id_or_link, list):
```

Comment:
We can certainly accept tuples and treat them the same as a list.  Since sets are unordered though, returning a list of the descriptions as we do now may not make sense.

We could have another path that accepts sets and then returns a dict with keys = set elements and vals set to the description.

What do you think?